crates/core/src/protocols/evalcheck/subclaims.rs

```diff
@@ -183,6 +208,44 @@ where
 	Ok(())
 }
 
+#[allow(clippy::too_many_arguments)]
+pub fn process_composite_sumcheck<U, F>(
+	comp: &CompositeMLE<F>,
+	meta: ProjectedBivariateMeta,
+	eval_point: &[F],
+	eval: F,
+	witness_index: &mut MultilinearExtensionIndex<U, F>,
+	constraint_builders: &mut Vec<ConstraintSetBuilder<F>>,
+	backend: &impl ComputationBackend,
+	projected: MultilinearExtension<PackedType<U, F>>,
+) -> Result<(), Error>
+where
+	U: UnderlierType + PackScalar<F>,
+	F: TowerField,
+{
+	process_projected_bivariate_witness(
```

Comment:
You can just add poly to the witness without using process_projected_bivariate_witness, and you're right about parallelization. However, I think we can take data from memoized_queries since it's also tensor products.